### ISSUE TITLE
v.N not specified in conds for `build_callables`leads to incorrect energies

### ISSUE THREAD

**Issue by bocklund** [pycalphad/pycalphad#212]

**Comment by bocklund** [comment]
Example:


```python
from pycalphad import Database, equilibrium, binplot, variables as v
import pycalphad
print(pycalphad.__version__)
dbf = Database("/Users/brandon/Projects/pycalphad/examples/alzn_mey.tdb")
comps = ['AL', 'ZN', 'VA']
phases = ['LIQUID']
conds = {v.P: 101325, v.T: 800, v.X('AL'): 0.5}
# correct answer
print('correct answer')
print(equilibrium(dbf, comps, phases, conds))
from pycalphad.codegen.callables import build_callables
print('cbs')
cbs = build_callables(dbf, comps, phases, conds={v.T: 1000, v.P: 101325})
print(equilibrium(dbf, comps, phases, conds, callables=cbs))
print('cbs2')
cbs2 = build_callables(dbf, comps, phases, conds={v.T: 1000, v.P: 101325, v.X('AL'): 0.01})
print(equilibrium(dbf, comps, phases, conds, callables=cbs2))
print('cbs3')
cbs3 = build_callables(dbf, comps, phases, conds={v.T: 1000, v.P: 101325, v.N: 1})
print(equilibrium(dbf, comps, phases, conds, callables=cbs3))
```

## Output

#### Correct answer, no callables
correct answer
```
<xarray.Dataset>
Dimensions:    (N: 1, P: 1, T: 1, X_AL: 1, component: 2, internal_dof: 2, vertex: 3)
Coordinates:
  * N          (N) float64 1.0
  * P          (P) float64 1.013e+05
  * T          (T) float64 800.0
  * X_AL       (X_AL) float64 0.5
  * vertex     (vertex) int64 0 1 2
  * component  (component) <U2 'AL' 'ZN'
Dimensions without coordinates: internal_dof
Data variables:
    NP         (N, P, T, X_AL, vertex) float64 1.0 nan nan
    GM         (N, P, T, X_AL) float64 -3.807e+04
    MU         (N, P, T, X_AL, component) float64 -3.131e+04 -4.482e+04
    X          (N, P, T, X_AL, vertex, component) float64 0.5 0.5 ... nan nan
    Y          (N, P, T, X_AL, vertex, internal_dof) float64 0.5 0.5 ... nan nan
    Phase      (N, P, T, X_AL, vertex) <U6 'LIQUID' '' ''
Attributes:
    engine:   pycalphad 0.7.1.post2+52.g4f64e482.dirty
    created:  2019-05-01T20:05:45.730185
```
#### callables=cbs (only T and P specififed)
cbs
```
<xarray.Dataset>
Dimensions:    (N: 1, P: 1, T: 1, X_AL: 1, component: 2, internal_dof: 2, vertex: 3)
Coordinates:
  * N          (N) float64 1.0
  * P          (P) float64 1.013e+05
  * T          (T) float64 800.0
  * X_AL       (X_AL) float64 0.5
  * vertex     (vertex) int64 0 1 2
  * component  (component) <U2 'AL' 'ZN'
Dimensions without coordinates: internal_dof
Data variables:
    NP         (N, P, T, X_AL, vertex) float64 1.0 nan nan
    GM         (N, P, T, X_AL) float64 5.628e+06
    MU         (N, P, T, X_AL, component) float64 5.631e+06 2.575e+05
    X          (N, P, T, X_AL, vertex, component) float64 0.9994 ... nan
    Y          (N, P, T, X_AL, vertex, internal_dof) float64 0.5 0.5 ... nan nan
    Phase      (N, P, T, X_AL, vertex) <U6 'LIQUID' '' ''
Attributes:
    engine:   pycalphad 0.7.1.post2+52.g4f64e482.dirty
    created:  2019-05-01T20:05:56.744699
```
#### callables=cbs2 (T, P, X specified)
cbs2
```
<xarray.Dataset>
Dimensions:    (N: 1, P: 1, T: 1, X_AL: 1, component: 2, internal_dof: 2, vertex: 3)
Coordinates:
  * N          (N) float64 1.0
  * P          (P) float64 1.013e+05
  * T          (T) float64 800.0
  * X_AL       (X_AL) float64 0.5
  * vertex     (vertex) int64 0 1 2
  * component  (component) <U2 'AL' 'ZN'
Dimensions without coordinates: internal_dof
Data variables:
    NP         (N, P, T, X_AL, vertex) float64 1.0 nan nan
    GM         (N, P, T, X_AL) float64 5.628e+06
    MU         (N, P, T, X_AL, component) float64 5.631e+06 2.575e+05
    X          (N, P, T, X_AL, vertex, component) float64 0.9994 ... nan
    Y          (N, P, T, X_AL, vertex, internal_dof) float64 0.5 0.5 ... nan nan
    Phase      (N, P, T, X_AL, vertex) <U6 'LIQUID' '' ''
Attributes:
    engine:   pycalphad 0.7.1.post2+52.g4f64e482.dirty
    created:  2019-05-01T20:05:56.945609
```
#### callables=cbs3(T,P,N specified)
cbs3
```
<xarray.Dataset>
Dimensions:    (N: 1, P: 1, T: 1, X_AL: 1, component: 2, internal_dof: 2, vertex: 3)
Coordinates:
  * N          (N) float64 1.0
  * P          (P) float64 1.013e+05
  * T          (T) float64 800.0
  * X_AL       (X_AL) float64 0.5
  * vertex     (vertex) int64 0 1 2
  * component  (component) <U2 'AL' 'ZN'
Dimensions without coordinates: internal_dof
Data variables:
    NP         (N, P, T, X_AL, vertex) float64 1.0 nan nan
    GM         (N, P, T, X_AL) float64 -3.807e+04
    MU         (N, P, T, X_AL, component) float64 -3.131e+04 -4.482e+04
    X          (N, P, T, X_AL, vertex, component) float64 0.5 0.5 ... nan nan
    Y          (N, P, T, X_AL, vertex, internal_dof) float64 0.5 0.5 ... nan nan
    Phase      (N, P, T, X_AL, vertex) <U6 'LIQUID' '' ''
Attributes:
    engine:   pycalphad 0.7.1.post2+52.g4f64e482.dirty
    created:  2019-05-01T20:05:57.019149
```

What's interesting is that it doesn't seem like the JIT is running again for the subsequent `build_callables` call. I think it has to do with the addition of N to the state_variables, which are used to construct the PhaseRecords

**Comment by bocklund** [comment]
Closed by #214
